File: Application/Channel/Channel.cs
using System;
using System.Collections.Generic;
using Application.Clip;
using Application.Game;

namespace Application.Channel
{
    public class Channel
    {
        public string GravatarProfileImage { get; set; }
        public DateTime CreatedAt { get; set; }
        public string Bio { get; set; }
        public string Twitter { get; set; }
        public string Instagram { get; set; }
        public string Youtube { get; set; }
        public string Twitch { get; set; }
        public ICollection<ClipDto> Clips { get; set; }
        public int OverallViews { get; set; }
        public string Username { get; set; }
        public ICollection<AllGamesDto> LikedGames { get; set; }
        public ICollection<ClipDto> LikedClips { get; set; }
        public int SubsrciberCount { get; set; }
        public bool SubscribedToUser { get; set; }
    }
}

File: client/src/infrastructure/models/channel.ts
﻿import { IClip } from "./clip";
import {IGame} from "./game";

export interface IChannel{
    gravatarProfileImage: string;
    createdAt: string;
    bio: string;
    instagram: string;
    youtube: string;
    twitch: string;
    twitter: string;
    clips: IClip[];
    username: string;
    overallViews: number;
    isUser: boolean;
    likedGames: IGame[];
    likedClips: IClip[];
    subsrciberCount: number;
    subscribedToUser: boolean;
}


export interface IPaginatedChannelResponse{
    pageNumber: number;
    pageSize: number;
    firstPage: string;
    lastPage: string;
    totalPages: number;
    totalRecords: number;
    nextPage: string;
    previousPage: string;
    data: IChannelUser[];
}

export interface IChannelUser{
    gravatarProfileImage: string;
    subscriberCount: number;
    username: string;
    subscribedToChannel: boolean;
}

export interface IChannelFormValues{
    bio?: string;
    twitter?: string;
    instagram?: string;
    twitch?: string;
    youtube?: string;
    username? : string;
}

export interface IChannelPasswordValues{
    oldPassword: string,
    newPassword: string;
    confirmPassword: string;
}